homeassistant/components/goalzero/__init__.py

```diff
@@ -88,10 +88,12 @@ def device_info(self):
             sw_version = self.api.data["firmwareVersion"]
         else:
             sw_version = None
+        try:
+            model
```

Comment:
How can this throw an exception?

Reply:
Whenever an entry is set up, then removed, then set up again it would throw an error `model` referenced before assignment.

Reply:
This was the best solution I could make without making `model` part of the instance.

Reply:
```python
    @property
    def device_info(self):
        """Return the device information of the entity."""
        info = {
            "identifiers": {(DOMAIN, self._server_unique_id)},
            "manufacturer": "Goal Zero",
            "name": self._name,
        }
        if self.api.sysdata:
            info["model"] = self.api.sysdata["model"]
        if self.api.data:
            info["sw_version"] = self.api.data["firmwareVersion"]
        return info
```